src/main/java/com/rakovets/course/java/core/practice/io/FileAnalyzeUtil.java

```diff
@@ -0,0 +1,200 @@
+    }
+
+    public static Map<Character, Integer> getQuantityOfCharsRepeating(Path filePath) {
+        Map<Character, Integer> usageOfChars = new HashMap<>();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(String.valueOf(filePath)))) {
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                line.toLowerCase().replaceAll("[^a-z]", "");
+                for (char currentChar : line.toCharArray()) {
+                    usageOfChars.put(currentChar, usageOfChars.getOrDefault(currentChar, 0) + 1);
+                }
+            }
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+        return usageOfChars;
+    }
+
+    public static Map<String, Integer> getQuantityOfWordsRepeating(Path filePath) {
+        Map<String, Integer> usageOfWords = new HashMap<>();
+        Map<String, Integer> sortedMap = new HashMap<>();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(String.valueOf(filePath)))) {
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                String text = line.toLowerCase().replaceAll("[\\[\\],.!?]", "").replaceAll("[\\s+]", " ");
+                for (String currentWord : text.split(" ")) {
+                    usageOfWords.put(currentWord, usageOfWords.getOrDefault(currentWord, 0) + 1);
+                }
+                usageOfWords.entrySet()
+                        .stream()
+                        .sorted(Collections.reverseOrder(Map.Entry.comparingByValue()))
+                        .forEachOrdered(x -> sortedMap.put(x.getKey(), x.getValue()));
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return sortedMap;
+    }
+
+    public static void getSortedNubersOrder(Path filePath) {
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(String.valueOf(filePath)))) {
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                String[] arrayOfNumbers = line.replaceAll("[^0-9\\s]", "").split(" ");
+                List<Integer> sortedList = Arrays.stream(arrayOfNumbers)
+                        .map(Integer::parseInt)
+                        .sorted()
+                        .collect(Collectors.toList());
+                try (BufferedWriter bufferedWriter =
+                             new BufferedWriter(new FileWriter(filePath + "_"))) {
+                    bufferedWriter.write(sortedList.toString());
+                    bufferedWriter.newLine();
+                    bufferedWriter.flush();
+                } catch (IOException e) {
+                    System.out.println(e.getMessage());
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+        public static Map<String, Double> getStudentsRate(Path filePath) {
```

Comment:
"Violation: There aren't correct indentations for inner code
Solution: Add correct indentations for inner code"